lib/SqlFormatter.php

```diff
@@ -214,38 +221,33 @@ protected static function getNextToken($string, $previous = null)
         }
 
         // Comment
-        if ($string[0] === '#' || (isset($string[1])&&($string[0]==='-'&&$string[1]==='-') || ($string[0]==='/'&&isset($string[1])&&$string[1]==='*'))) {
-            // Comment until end of line
-            if ($string[0] === '-' || $string[0] === '#') {
-                $last = strpos($string, "\n");
-                $type = self::TOKEN_TYPE_COMMENT;
-            } else { // Comment until closing comment tag or end of query
-                if (strpos($string, "*/", 2) !== false) {
-                    $last = strpos($string, "*/", 2) + 2;
-                } else {
-                    $last = strlen($string);
-                }
-                $type = self::TOKEN_TYPE_BLOCK_COMMENT;
-            }
-
-            if ($last === false) {
-                $last = strlen($string);
+        $found = false;
+        foreach (self::$comment_tokens as $comment) {
+            $start = $comment[0];
+            $end = isset($comment[1]) ? $comment[1] : "\n";
+            $found = substr($string, 0, strlen($start)) === $start;
+            if ($found) {
+                break;
             }
-
+        }
+        if ($found) {
```

Comment:
tohle by šlo dát přímo do toho foreache a nemusíš to kontrolovat ještě zvlášť

Reply:
https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Myslim si, ze je spravnejsie snazit sa mat co najmenej urovni kodu. Je to tak lachsie citatelne a menej nachylne na chyby.
V prvej casti sa zisti ci je to coment. V druhej casti sa to spracuje.

Reply:
`Myslim si, ze je spravnejsie snazit sa mat co najmenej urovni kodu` - souhlasím, ale když už tam ten if je :)
`Je to tak lachsie citatelne a menej nachylne na chyby.` - v pohodě já nejsem proti :)